autofit/optimize/non_linear/nested_sampling/dynesty.py

```diff
@@ -139,77 +139,105 @@ def _simple_fit(self, model: AbstractPriorModel, fitness_function) -> Result:
             The model which generates instances for different points in parameter space. This maps the points from unit
             cube values to physical values via the priors.
         fitness_function
-            A function that fits this model to the data, returning the likelihood of the fit.
+            A function that fits this model to the data, returning the log likelihood of the fit.
 
         Returns
         -------
-        A result object comprising the best-fit model instance, likelihood and an *Output* class that enables analysis
+        A result object comprising the best-fit model instance, log_likelihood and an *Output* class that enables analysis
         of the full chains used by the fit.
         """
-        dynesty_output = DynestyOutput(model=model, paths=self.paths)
 
         if os.path.exists("{}/{}.pickle".format(self.paths.chains_path, "dynesty")):
 
-            with open(
-                "{}/{}.pickle".format(self.paths.chains_path, "dynesty"), "rb"
-            ) as f:
-                dynesty_sampler = pickle.load(f)
+            sampler = self.load_sampler
 
         else:
 
-            dynesty_sampler = self.sampler_fom_model_and_fitness(
+            sampler = self.sampler_fom_model_and_fitness(
                 model=model, fitness_function=fitness_function
             )
 
         # These hacks are necessary to be able to pickle the sampler.
 
-        dynesty_sampler.rstate = np.random
+        sampler.rstate = np.random
         pool = Pool(processes=1)
-        dynesty_sampler.pool = pool
-        dynesty_sampler.M = pool.map
+        sampler.pool = pool
+        sampler.M = pool.map
 
         dynesty_finished = False
 
         while dynesty_finished is False:
 
             try:
-                iterations_before_run = np.sum(dynesty_sampler.results.ncall)
+                iterations_before_run = np.sum(sampler.results.ncall)
             except AttributeError:
                 iterations_before_run = 0
 
-            dynesty_sampler.run_nested(maxcall=self.iterations_per_update)
+            sampler.run_nested(maxcall=self.iterations_per_update)
 
-            iterations_after_run = np.sum(dynesty_sampler.results.ncall)
+            iterations_after_run = np.sum(sampler.results.ncall)
 
             with open(
                 f"{self.paths.chains_path}/dynesty.pickle", "wb"
             ) as f:
-                pickle.dump(dynesty_sampler, f)
+                pickle.dump(sampler, f)
 
             if iterations_before_run == iterations_after_run:
                 dynesty_finished = True
```

Comment:
So this is where you'd use a break statement.

You only need a boolean to break the loop to implement more complicated behaviours. Something like cleaning up before the loop closes.